control-plane/agents/core/src/volume/service.rs

```diff
@@ -21,21 +21,14 @@ impl Service {
     /// Get volumes
     #[tracing::instrument(level = "debug", err)]
     pub(super) async fn get_volumes(&self, request: &GetVolumes) -> Result<Volumes, SvcError> {
-        let volumes = self.registry.get_volumes_status().await;
+        let volumes = self.registry.get_volumes().await;
 
-        let volumes = match &request.filter {
+        // The filter criteria is matched against the volume state.
+        let filtered_volumes = match &request.filter {
             Filter::None => volumes,
-            Filter::NodeVolume(node, volume) => volumes
+            Filter::Volume(volume_id) => volumes
                 .iter()
-                .filter(|volume_iter| {
-                    volume_iter.children.iter().any(|c| &c.node == node)
-                        && &volume_iter.uuid == volume
-                })
-                .cloned()
-                .collect(),
-            Filter::Volume(volume) => volumes
-                .iter()
-                .filter(|volume_iter| &volume_iter.uuid == volume)
+                .filter(|v| &v.get_spec().uuid == volume_id)
```

Comment:
Could we do get_volume_state(volume_id) instead to get the volumeNotFound error?

Reply:
Done.